pygments/lexer.py

```diff
@@ -255,6 +255,8 @@ def get_tokens(self, text, unfiltered=False):
         if self.ensurenl and not text.endswith('\n'):
             text += '\n'
 
+        self._input_for_tests = text
```

Comment:
This looks kludgy, is there a way to avoid storing the input indefinitely in production?